src/System.Net.Http/src/System/Net/Http/Unix/CurlTransportContext.cs

```diff
@@ -2,33 +2,43 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Security.Authentication.ExtendedProtection;
+using System.Threading;
 
 namespace System.Net.Http
 {
-    internal class CurlTransportContext : TransportContext
+    internal sealed class CurlTransportContext : TransportContext
     {
-        private CurlChannelBinding _channelBinding = new CurlChannelBinding();
+        private CurlChannelBinding _channelBinding;
 
         internal CurlChannelBinding CurlChannelBinding
         {
             get
             {
-                return _channelBinding;
+                CurlChannelBinding binding = Volatile.Read(ref _channelBinding);
+                if (binding == null)
+                {
+                    var newBinding = new CurlChannelBinding();
+                    binding = Interlocked.CompareExchange(ref _channelBinding, newBinding, null);
```

Comment:
> I assume you're doing all this Interlocked stuff because you are trying to protect against multi-threaded access to the GetChannelBinding method?

Yeah.  I was also slightly concerned (though maybe inappropriately?) that GetChannelBinding might be used concurrently with the SSL handshake happening and the token getting set into the ChannelBinding, which would also force it into existence.